pkg/skaffold/render/generate/generate.go

```diff
@@ -26,75 +28,69 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy/kustomize"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/event"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/instrumentation"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/kubernetes/manifest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/render/kptfile"
 	latestV2 "github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest/v2"
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 )
 
 // NewGenerator instantiates a Generator object.
-func NewGenerator(workingDir string, config latestV2.Generate) *Generator {
+func NewGenerator(workingDir string, config latestV2.Generate, hydrationDir string) *Generator {
 	return &Generator{
-		workingDir: workingDir,
-		config:     config,
+		workingDir:   workingDir,
+		hydrationDir: hydrationDir,
+		config:       config,
 	}
 }
 
 // Generator provides the functions for the manifest sources (raw manifests, helm charts, kustomize configs and remote packages).
 type Generator struct {
-	workingDir string
-	config     latestV2.Generate
+	workingDir   string
+	hydrationDir string
+	config       latestV2.Generate
 }
 
-// Generate parses the config resources from the paths in .Generate.Manifests. This path can be the path to raw manifest,
-// kustomize manifests, helm charts or kpt function configs. All should be file-watched.
-func (g *Generator) Generate(ctx context.Context) (manifest.ManifestList, error) {
-	// exclude remote url.
-	var paths []string
-	// TODO(yuwenma): Apply new UX, kustomize kpt and helm
-	for _, path := range g.config.RawK8s {
+func excludeRemote(paths []string) []string {
```

Comment:
how will remote paths be handled? do we not want to support generating from remote source?

Reply:
We will. See the TODO comments I added below. It will be handled in separate PRs